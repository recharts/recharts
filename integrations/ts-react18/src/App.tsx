import './App.css';
import { Bar, BarChart, Legend, Tooltip, XAxis, YAxis } from 'recharts';

function App() {
  return (
    <>
      <h1>Vite + React 18 + TypeScript + Recharts</h1>
      <BarChart
        width={600}
        height={300}
        data={[
          { name: 'A', value: 100 },
          { name: 'B', value: 200 },
        ]}
      >
        <XAxis dataKey="name" />
        <YAxis />
        <Bar dataKey="value" fill="gold" name="Tooltip value" />
        <Legend />
        <Tooltip />
      </BarChart>
    </>
  );
}

export default App;
