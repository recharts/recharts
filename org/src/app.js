/* eslint-disable no-underscore-dangle */
import React from 'react';
import ReactDOM from 'react-dom';
import { applyRouterMiddleware, browserHistory } from 'react-router';
import { syncHistoryWithStore } from 'react-router-redux';
import { useScroll } from 'react-router-scroll';
import makeRoutes from './routes';
import Root from './containers/Root';
import configureStore from './redux/configureStore';

const initialState = window.__INITIAL_STATE__;
const store = configureStore(initialState, browserHistory);
const history = syncHistoryWithStore(browserHistory, store, {
  selectLocationState: (state) => state.router,
});

const routes = makeRoutes(store);

ReactDOM.render(
  <Root history={history} routes={routes} store={store} render={applyRouterMiddleware(useScroll())} />,
  document.getElementById('app'),
);
