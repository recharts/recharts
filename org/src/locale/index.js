import zhCN from './zh-CN';
import enUS from './en-US';

export default {
  'zh-CN': zhCN,
  'en-US': enUS,
};
