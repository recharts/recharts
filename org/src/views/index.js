import IndexView from './IndexView';
import GuideView from './GuideView';
import APIView from './APIView';
import ExamplesView from './ExamplesView';
import BlogView from './BlogView';
import Storybook from './Storybook';

export { IndexView, GuideView, APIView, ExamplesView, BlogView, Storybook };
