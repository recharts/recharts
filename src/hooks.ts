import type { AxisId } from './state/axisMapSlice';
import { BaseAxisWithScale, selectAxisWithScale } from './state/selectors/axisSelectors';
import { useAppSelector } from './state/hooks';

export const useXAxis = (xAxisId: AxisId): BaseAxisWithScale | undefined =>
  useAppSelector(state => selectAxisWithScale(state, 'xAxis', xAxisId));

export const useYAxis = (yAxisId: AxisId): BaseAxisWithScale | undefined =>
  useAppSelector(state => selectAxisWithScale(state, 'yAxis', yAxisId));
