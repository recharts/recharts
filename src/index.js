import 'ima.js-babel6-polyfill';
import 'core-js/es6/math';
import Surface from './container/Surface';

import CartesianAxis from './component/CartesianAxis';
import CartesianGrid from './component/CartesianGrid';
import Legend from './component/Legend';
import Tooltip from './component/Tooltip';
import PolarGrid from './component/PolarGrid';
import PolarRadiusAxis from './component/PolarRadiusAxis';
import PolarAngleAxis from './component/PolarAngleAxis';
import Brush from './component/Brush';
import ReferenceLine from './component/ReferenceLine';

import Sector from './shape/Sector';
import Curve from './shape/Curve';
import Rectangle from './shape/Rectangle';
import Triangle from './shape/Triangle';
import Polygon from './shape/Polygon';
import Dot from './shape/Dot';

import LineChart from './chartWrapper/LineChart';
import Line from './chart/Line';

import BarChart from './chartWrapper/BarChart';
import Bar from './chart/Bar';

import PieChart from './chartWrapper/PieChart';
import Pie from './chart/Pie';

import TreemapChart from './chartWrapper/TreemapChart';

import RadarChart from './chartWrapper/RadarChart';
import Radar from './chart/Radar';

import ScatterChart from './chartWrapper/ScatterChart';
import Scatter from './chart/Scatter';

import AreaChart from './chartWrapper/AreaChart';
import Area from './chart/Area';

import RadialBarChart from './chartWrapper/RadialBarChart';
import RadialBar from './chart/RadialBar';

import ComposedChart from './chartWrapper/ComposedChart';

import XAxis from './chart/XAxis';
import YAxis from './chart/YAxis';
import ZAxis from './chart/ZAxis';

export {
  // container
  Surface,

  // chart
  Radar,

  // componnet
  CartesianAxis, CartesianGrid,
  Legend, Tooltip, Brush, ReferenceLine,
  PolarGrid, PolarRadiusAxis, PolarAngleAxis,

  // shape
  Sector, Curve, Rectangle, Triangle, Polygon, Dot,

  LineChart, Line,
  AreaChart, Area,
  BarChart, Bar,
  PieChart, Pie,
  ScatterChart, Scatter,
  RadialBarChart, RadialBar,
  TreemapChart,
  RadarChart, Radar,
  ComposedChart,

  XAxis, YAxis, ZAxis,
};
