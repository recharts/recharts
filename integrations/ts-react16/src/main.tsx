import { StrictMode } from 'react';
import { render } from 'react-dom';
import './index.css';
import App from './App.tsx';

render(
  <StrictMode>
    <App />
  </StrictMode>,
  document.getElementById('root')!,
);
