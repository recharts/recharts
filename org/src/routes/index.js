import React, { Component } from 'react';
import { Route, IndexRoute, Redirect, IndexRedirect } from 'react-router';
import _ from 'lodash';
import { IndexView, GuideView, APIView, ExamplesView, BlogView, Storybook } from '../views';
import Frame from '../layouts/Frame';
import Locale from '../locale';

const locales = Object.keys(Locale);

const hasLocale = (pathname) => {
  for (let i = 0, len = locales.length; i < len; i++) {
    if (pathname.indexOf(locales[i]) >= 0) {
      return true;
    }
  }

  return false;
};

const onEnter = (nextState, replace) => {
  const pathname = _.get(nextState, 'location.pathname', '/');

  if (!hasLocale(pathname)) {
    replace({
      pathname: `/en-US${pathname}`,
      query: _.get(nextState, 'location.query', {}),
    });
  }
};

export default (store) => (
  <Route path="/" component={Frame} onEnter={onEnter}>
    <IndexRedirect to="/en-US" />
    <Route path="/*/guide(/:name)" component={GuideView} />
    <Route path="/*/api(/:name)" component={APIView} />
    <Route path="/*/examples(/:name)" component={ExamplesView} />
    <Route path="/*/blog" component={BlogView} />
    <Route path="/*/storybook" component={Storybook} />
    <Route path="*" component={IndexView} />
  </Route>
);
