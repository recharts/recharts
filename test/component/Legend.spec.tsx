import React from 'react';
import { render } from '@testing-library/react';
import { Legend, LineChart, Line } from '../../src';

describe('<Legend />', () => {
  const data = [
    { value: 'Apple', color: '#ff7300' },
    { value: 'Samsung', color: '#bb7300' },
    { value: 'Huawei', color: '#887300' },
    { value: 'Sony', color: '#667300' },
  ];

  test('Render 4 legend items in simple Legend', () => {
    const { container } = render(<Legend width={500} height={30} payload={data} />);

    expect(container.querySelectorAll('.recharts-default-legend')).toHaveLength(1);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item')).toHaveLength(4);
  });

  test('Render customized legend when content is set to be a react element', () => {
    const CustomizedLegend = () => <div className="customized-legend">test</div>;
    const { container } = render(<Legend width={500} height={30} payload={data} content={<CustomizedLegend />} />);

    expect(container.querySelectorAll('.recharts-default-legend')).toHaveLength(0);
    expect(container.querySelectorAll('.customized-legend')).toHaveLength(1);
  });

  test('Does not render items with a type of `none`', () => {
    const dataWithNone = [
      { value: 'Apple', color: '#ff7300' },
      { value: 'Samsung', color: '#bb7300' },
      { value: 'Huawei', color: '#887300' },
      { value: 'Sony', type: 'none' },
    ];
    const { container } = render(<Legend width={500} height={30} payload={dataWithNone as any} />);

    expect(container.querySelectorAll('.recharts-default-legend')).toHaveLength(1);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item')).toHaveLength(3);
  });

  test('Renders `strokeDasharray` (if present) in Legend when iconType is set to `plainline`', () => {
    const { container } = render(
      <LineChart width={600} height={300} data={data} margin={{ top: 5, right: 30, left: 20, bottom: 5 }}>
        <Legend iconType="plainline" />
        <Line type="monotone" dataKey="pv" stroke="#8884d8" activeDot={{ r: 8 }} strokeDasharray="5 5" />
        <Line type="monotone" dataKey="uv" stroke="#82ca9d" />
      </LineChart>,
    );

    expect(container.querySelectorAll('.recharts-default-legend')).toHaveLength(1);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item')).toHaveLength(2);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item path')).toHaveLength(0);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item line')).toHaveLength(2);
  });

  test('Does not render `strokeDasharray` (if not present) in Legend when iconType is set to something else than `plainline`', () => {
    const { container } = render(
      <LineChart width={600} height={300} data={data}>
        <Legend iconType="line" />
        <Line dataKey="pv" />
        <Line type="monotone" dataKey="uv" stroke="#82ca9d" />
      </LineChart>,
    );

    expect(container.querySelectorAll('.recharts-default-legend')).toHaveLength(1);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item')).toHaveLength(2);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item path')).toHaveLength(2);
    expect(container.querySelectorAll('.recharts-default-legend .recharts-legend-item line')).toHaveLength(0);
  });
});
