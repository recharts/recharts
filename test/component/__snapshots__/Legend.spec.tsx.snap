// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Legend /> > content as a React Component > should pass parameters to the Component 1`] = `
{
  "align": "center",
  "chartHeight": 300,
  "chartWidth": 600,
  "content": [Function],
  "iconSize": 14,
  "layout": "horizontal",
  "margin": {
    "bottom": 5,
    "left": 20,
    "right": 30,
    "top": 5,
  },
  "onBBoxUpdate": [Function],
  "payload": [
    {
      "color": "#8884d8",
      "dataKey": "pv",
      "inactive": false,
      "payload": {
        "activeDot": {
          "r": 8,
        },
        "animateNewValues": true,
        "animationBegin": 0,
        "animationDuration": 1500,
        "animationEasing": "ease",
        "connectNulls": false,
        "dataKey": "pv",
        "dot": true,
        "fill": "#fff",
        "hide": false,
        "isAnimationActive": true,
        "label": false,
        "legendType": "line",
        "points": [],
        "stroke": "#8884d8",
        "strokeDasharray": "5 5",
        "strokeWidth": 1,
        "type": "monotone",
        "xAxisId": 0,
        "yAxisId": 0,
      },
      "type": "line",
      "value": "pv",
    },
    {
      "color": "#82ca9d",
      "dataKey": "uv",
      "inactive": false,
      "payload": {
        "activeDot": true,
        "animateNewValues": true,
        "animationBegin": 0,
        "animationDuration": 1500,
        "animationEasing": "ease",
        "connectNulls": false,
        "dataKey": "uv",
        "dot": true,
        "fill": "#fff",
        "hide": false,
        "isAnimationActive": true,
        "label": false,
        "legendType": "line",
        "points": [],
        "stroke": "#82ca9d",
        "strokeWidth": 1,
        "type": "monotone",
        "xAxisId": 0,
        "yAxisId": 0,
      },
      "type": "line",
      "value": "uv",
    },
  ],
  "verticalAlign": "bottom",
  "width": 550,
}
`;

exports[`<Legend /> > content as a function > should pass parameters to the function 1`] = `
{
  "align": "center",
  "chartHeight": 300,
  "chartWidth": 600,
  "content": [Function],
  "iconSize": 14,
  "layout": "horizontal",
  "margin": {
    "bottom": 5,
    "left": 20,
    "right": 30,
    "top": 5,
  },
  "onBBoxUpdate": [Function],
  "payload": [
    {
      "color": "#8884d8",
      "dataKey": "pv",
      "inactive": false,
      "payload": {
        "activeDot": {
          "r": 8,
        },
        "animateNewValues": true,
        "animationBegin": 0,
        "animationDuration": 1500,
        "animationEasing": "ease",
        "connectNulls": false,
        "dataKey": "pv",
        "dot": true,
        "fill": "#fff",
        "hide": false,
        "isAnimationActive": true,
        "label": false,
        "legendType": "line",
        "points": [],
        "stroke": "#8884d8",
        "strokeDasharray": "5 5",
        "strokeWidth": 1,
        "type": "monotone",
        "xAxisId": 0,
        "yAxisId": 0,
      },
      "type": "line",
      "value": "pv",
    },
    {
      "color": "#82ca9d",
      "dataKey": "uv",
      "inactive": false,
      "payload": {
        "activeDot": true,
        "animateNewValues": true,
        "animationBegin": 0,
        "animationDuration": 1500,
        "animationEasing": "ease",
        "connectNulls": false,
        "dataKey": "uv",
        "dot": true,
        "fill": "#fff",
        "hide": false,
        "isAnimationActive": true,
        "label": false,
        "legendType": "line",
        "points": [],
        "stroke": "#82ca9d",
        "strokeWidth": 1,
        "type": "monotone",
        "xAxisId": 0,
        "yAxisId": 0,
      },
      "type": "line",
      "value": "uv",
    },
  ],
  "verticalAlign": "bottom",
  "width": 550,
}
`;
